`Reskin` |  DAO Hierarchy level indication

Da-Colon opened the pull request:
**Description:**

- Implemented the new Figma design to clearly indicate the currently viewed DAO.

**Notes:**

1. **Code Cleanup:**
   - The `NodeInfoCard` contains a significant amount of commented-out code related to the old formatting, which can likely be removed for clarity.

2. **Component Update:**
   - The DAO Info Card on the DAO Dashboard and Hierarchy pages, which previously used the same components, are now quite different in the new design (e.g., no gear icon, reduced icon size). We should decouple these components and create a separate component for the Hierarchy page rather than modifying the existing one.

![Hierarchy Page](https://github.com/decentdao/fractal-interface/assets/38386583/29e972ef-b08b-4f60-90d0-14f9a0f86a20)

---

**Additional Information:**

- This branch is based on my GraphQL bug fix branch to allow for proper testing and visibility.


Closes #1720

xraystyle1980 commented:
@Da-Colon updated the [Hierarchy handoff to include a prototype](https://www.figma.com/proto/9hUYtvCKjOyB92vt8ukYtD/Product-Handoff?page-id=122%3A13462&node-id=2746-5942&viewport=1037%2C-346%2C0.35&t=57SJEp8vmUSy3SkE-1&scaling=scale-down&starting-point-node-id=2746%3A5942&show-proto-sidebar=1) to show clicking on the full card should switch organizations. 

*Also includes hover and active states for the hierarchy cards. 

@Da-Colon I added the three dots icon-button to these designs. I can't recall if we removed them intentionally or they just didn't make it to the reskin design. @tomstuart123 can you weigh in here?

Da-Colon commented:
> @Da-Colon updated the [Hierarchy handoff to include a prototype](https://www.figma.com/proto/9hUYtvCKjOyB92vt8ukYtD/Product-Handoff?page-id=122%3A13462&node-id=2746-5942&viewport=1037%2C-346%2C0.35&t=57SJEp8vmUSy3SkE-1&scaling=scale-down&starting-point-node-id=2746%3A5942&show-proto-sidebar=1) to show clicking on the full card should switch organizations.
> 
> *Also includes hover and active states for the hierarchy cards.
> 
> @Da-Colon I added the three dots icon-button to these designs. I can't recall if we removed them intentionally or they just didn't make it to the reskin design. @tomstuart123 can you weigh in here?

I'm going to take this comment and make a new Issue. Between Adam's PR #1766 and the work I saw him doing earlier today. There will be alot of code changes coming for the `DAOInfoCard` which I'll need to touch to adjust the settings icon and the hover/active states. 

We can talk on Monday, but I think it might be better to decouple the use of this component between two pages and have separate components. This will make this alot easier to follow code-wise I believe. 

But I believe this PR covers the single issue indication of the Current DAO.

adamgall commented on src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx:
```diff
@@ -68,31 +72,42 @@ export function DaoHierarchyNode({
-        guardContracts={daoData?.freezeGuardContracts}
-      />
+      <Box
+        border={currenDAOAddress === getAddress(daoAddress || '') ? '4px solid' : 'none'}
+        borderColor={currenDAOAddress === getAddress(daoAddress || '') ? 'neutral-4' : 'none'}
```
`getAddress` will throw if you give it the empty string. Don't do this plz

adamgall commented on src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx:
```diff
@@ -68,31 +72,42 @@ export function DaoHierarchyNode({
       {fractalNode?.nodeHierarchy.childNodes.map(childNode => (
-        <HStack
+        <Flex
+          flexDirection="row"
```
Row is default don't need to specify it

adamgall commented on src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx:
```diff
@@ -25,6 +26,9 @@ export function DaoHierarchyNode({
   depth: number;
 }) {
+  const {
+    node: { daoAddress: currenDAOAddress },
```
Nit: typo here: `currentDAOAddress`

adamgall commented on src/components/pages/DaoHierarchy/DaoHierarchyNode.tsx:
```diff
@@ -68,31 +72,42 @@ export function DaoHierarchyNode({
             boxSize="32px"
-            color="neutral-6"
+            color={
+              currenDAOAddress === getAddress(childNode.daoAddress || '') ? 'celery-0' : 'neutral-6'
```
No empty string into `getAddress` plz

Da-Colon replied:
o dang didn't know that. I'll get this fixed then

DarksightKellar reviewed (changes requested):
Changes LGTM, but should the border not have a radius on it? There's a visible gap:

<img width="224" alt="Screenshot 2024-05-13 at 12 42 22" src="https://github.com/decentdao/fractal-interface/assets/7101382/cdf207c4-89b2-408e-81d2-e446dee57532">

xraystyle1980 reviewed (changes requested):
Border radius on active card is missing

tomstuart123 commented:
Extra request - how hard would it be to only include the border once the DAO has loaded?

Reason - see below. The border jumps around for a bit as the DAO in the hierarchy load animation forces it to reverberate
![Screenshot 2024-05-13 at 15 59 40](https://github.com/decentdao/fractal-interface/assets/31554133/3cc788b5-e944-4d4a-aa68-1e464f112cd2)

link to try it - https://deploy-preview-1767.app.dev.fractalframework.xyz/hierarchy?dao=sep:0x065D37327801F5a0c01AD7CfB388288eb2378fFf

Da-Colon commented:
Good catch @DarksightKellar 

![localhost_3000_hierarchy_dao=sep_0x35b798b19F70b236b9Dcb156F21F4f0f4Ca44C49(Desktop)](https://github.com/decentdao/fractal-interface/assets/38386583/6f447881-4fc8-4bbd-97ea-63c5e612b183)


@tomstuart123 I was able to make it happen

![Node loading](https://github.com/decentdao/fractal-interface/assets/38386583/3e91d33f-d264-4cb2-9486-176a30607457)

xraystyle1980 reviewed (dismissed):
The Favorite button is super tiny. And can you make the entire card clickable? [See prototype](https://www.figma.com/proto/9hUYtvCKjOyB92vt8ukYtD/Product-Handoff?page-id=122%3A13462&node-id=2746-5942&viewport=719%2C-376%2C0.32&t=6P9F7fu4rIVHJ5X8-1&scaling=scale-down&starting-point-node-id=2746%3A5942&show-proto-sidebar=1)

tomstuart123 commented:
@Da-Colon - that looks fire, Thanks so much

@xraystyle1980 from my understanding this is linked to @Da-Colon 's comment above. As this component and DAO home have the same PR, the tiny Favourite's icon is shared between them. This should get resolved by an existing PR on the tiny 'Favourites' on dao home. 

Is that correct devs?

Da-Colon commented:
> @Da-Colon - that looks fire, Thanks so much
> 
> @xraystyle1980 from my understanding this is linked to @Da-Colon 's comment above. As this component and DAO home have the same PR, the tiny Favourite's icon is shared between them. This should get resolved by an existing PR on the tiny 'Favourites' on dao home.
> 
> Is that correct devs?



> @Da-Colon - that looks fire, Thanks so much
> 
> @xraystyle1980 from my understanding this is linked to @Da-Colon 's comment above. As this component and DAO home have the same PR, the tiny Favourite's icon is shared between them. This should get resolved by an existing PR on the tiny 'Favourites' on dao home.
> 
> Is that correct devs?

Yep thats correct. @xraystyle1980 To another point. I think the two components have been separated already and only have shared components (such as the `FavoriteIcon` component. Which I think is why the full click isn't working. I'll fix the whole linking in the follow PR for issue #1773

DarksightKellar reviewed (changes requested):
Lol sorry to be that guy, but there's still a gap when you look closely 😅
<img width="125" alt="Screenshot 2024-05-14 at 15 54 53" src="https://github.com/decentdao/fractal-interface/assets/7101382/99be3712-a8c6-40f5-9133-ee24558369c0">

I think I've had this problem before on a component that was too small to notice, where even matching radii didn't seem to work visually.

Da-Colon commented:
> Lol sorry to be that guy, but there's still a gap when you look closely 😅 <img alt="Screenshot 2024-05-14 at 15 54 53" width="125" src="https://private-user-images.githubusercontent.com/7101382/330486567-99be3712-a8c6-40f5-9133-ee24558369c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU2OTk3MTYsIm5iZiI6MTcxNTY5OTQxNiwicGF0aCI6Ii83MTAxMzgyLzMzMDQ4NjU2Ny05OWJlMzcxMi1hOGM2LTQwZjUtOTEzMy1lZTI0NTU4MzY5YzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUxNCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MTRUMTUxMDE2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NDM3ZjIxM2E0ZDYwNmQwMzk3ODJiYzdjZjJlNzAwYWNmYjJmMGM1N2FmMjFhMzIwYmQ2NGQ4ZGRiOGNlMGVkMyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.mSQ-3x6h7-8ngW4MpNhkii4LhHWaH-wSu_pIAy0Nynw">
> 
> I think I've had this problem before on a component that was too small to notice, where even matching radii didn't seem to work visually.
> 
> EDIT: Hmm, maybe reducing the radius on the card itself so the border occludes the curve (only on active state) might do the trick? I have a feeling it'll lead to unwanted jank tho

ah blah, good eye. yeah I need to just move this all down into the DAONodeInfoCard, which would be the plan. anything else I'd do here would just be hacky. Can we get this merged in as is and I work on cleaning this up in the next PR?
